### PR TITLE
Update third_party_environments with Gym-Gridworld repo

### PR DESCRIPTION
**Pull request by sparisi**
Added Gym-Gridworlds.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Documentation only change (no code changed)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Screenshots

Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Kallinteris-Andreas** [comment]
Environments and environment categories are alphabetically sorted
update your PR name

**Review by Kallinteris-Andreas — APPROVED** [review]

File: docs/environments/third_party_environments.md [hunk 1]
```diff
@@ -285,7 +292,6 @@ goal-RL ([Gymnasium-Robotics](https://robotics.farama.org/)).
 
   SimpleGrid is a super simple and minimal grid environment for Gymnasium. It is easy to use and customise and it is intended to offer an environment for rapidly testing and prototyping different RL algorithms.
 
-
```
Comment (Kallinteris-Andreas) [inline docs/environments/third_party_environments.md:288]: why was this line removed?

Comment (sparisi) [reply docs/environments/third_party_environments.md]: I accidentally cut it when I cut-pasted the description of my environment, sorry. Reverted.
